esv_reference_server/handlers_headers.py

```diff
@@ -84,16 +87,50 @@ async def get_headers_by_height(request: web.Request) -> web.Response:
         return web.HTTPServiceUnavailable()
 
 
+def _convert_json_tips_to_binary(result: List[HeaderSVTip]) -> bytearray:
+    headers_array = bytearray()
+    for tip in result:
+        version = tip['header']['version']
+        prev_hash = hex_str_to_hash(tip['header']['prevBlockHash'])
+        merkle_root = hex_str_to_hash(tip['header']['merkleRoot'])
+        timestamp = tip['header']['creationTimestamp']
+        target = tip['header']['difficultyTarget']
+        nonce = tip['header']['nonce']
+
+        raw_header = pack_header(version, prev_hash, merkle_root, timestamp, target, nonce)
+        block_hash = double_sha256(raw_header)
+        assert hash_to_hex_str(block_hash) == tip['header']['hash']
+        headers_array += raw_header
+        headers_array += struct.pack("<I", tip['height'])
+    return headers_array
+
+
 async def get_chain_tips(request: web.Request) -> web.Response:
     client_session: aiohttp.ClientSession = request.app['client_session']
     app_state: ApplicationState = request.app['app_state']
+    accept_type = request.headers.get('Accept', 'application/json')
+
+    params = request.rel_url.query
+    longest_chain = params.get('longest_chain', '0')
+
     try:
         url_to_fetch = f"{app_state.header_sv_url}/api/v1/chain/tips"
         request_headers = {'Accept': 'application/json'}
         async with client_session.get(url_to_fetch, headers=request_headers) as response:
             result: List[HeaderSVTip] = await response.json()
         response_headers = {'User-Agent': 'ESV-Ref-Server'}
+
+        if longest_chain == '1':
+            for tip in result:
+                if tip['state'] == 'LONGEST_CHAIN':
+                    result = [tip]
```

Comment:
You are altering a structure you are iterating over. You should break here.

Reply:
Thanks will fix